From 1f36aabfe81aa168da11229158faf70b68b9dfd7 Mon Sep 17 00:00:00 2001
From: Mikael Pettersson <mikpelinux@gmail.com>
Date: Mon, 28 Jul 2025 13:44:46 +0200
Subject: [PATCH] restore bootstrap with --enable-checking=release [PR121260]

Current trunk doesn't bootstrap with --enable-checking=release
due to improper nesting of namespaces and #if CHECKING_P blocks.
This corrects that.

gcc/
	PR other/121260
	* diagnostics/changes.cc: Correct nesting of namespaces
	and #if CHECKING_P blocks.
	* diagnostics/context.cc: Likewise.
	* diagnostics/html-sink.cc: Likewise.
	* diagnostics/output-spec.cc: Likewise.
	* diagnostics/sarif-sink.cc: Likewise.

Signed-off-by: Mikael Pettersson <mikpelinux@gmail.com>
Signed-off-by: David Malcolm <dmalcolm@redhat.com>
---
 gcc/diagnostics/changes.cc     | 8 +++++++-
 gcc/diagnostics/context.cc     | 3 ++-
 gcc/diagnostics/html-sink.cc   | 3 ++-
 gcc/diagnostics/output-spec.cc | 3 ++-
 gcc/diagnostics/sarif-sink.cc  | 3 ++-
 5 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/gcc/diagnostics/changes.cc b/gcc/diagnostics/changes.cc
index 290d6022b5ae..e1caab0e59fc 100644
--- a/gcc/diagnostics/changes.cc
+++ b/gcc/diagnostics/changes.cc
@@ -1850,8 +1850,13 @@ run_all_tests ()
 }
 
 } // namespace diagnostics::changes::selftest
+
+#endif /* CHECKING_P */
+
 } // namespace diagnostics::changes
 
+#if CHECKING_P
+
 namespace selftest { // diagnostics::selftest
 
 /* Run all of the selftests within this file.  */
@@ -1863,6 +1868,7 @@ changes_cc_tests ()
 }
 
 } // namespace selftest
-} // namespace diagnostics
 
 #endif /* CHECKING_P */
+
+} // namespace diagnostics
diff --git a/gcc/diagnostics/context.cc b/gcc/diagnostics/context.cc
index 0dbc14810395..85f7d2a357ac 100644
--- a/gcc/diagnostics/context.cc
+++ b/gcc/diagnostics/context.cc
@@ -2130,10 +2130,11 @@ context_cc_tests ()
 }
 
 } // namespace diagnostics::selftest
-} // namespace diagnostics
 
 #endif /* #if CHECKING_P */
 
+} // namespace diagnostics
+
 #if __GNUC__ >= 10
 #  pragma GCC diagnostic pop
 #endif
diff --git a/gcc/diagnostics/html-sink.cc b/gcc/diagnostics/html-sink.cc
index 07e71871d181..13d6309b7a45 100644
--- a/gcc/diagnostics/html-sink.cc
+++ b/gcc/diagnostics/html-sink.cc
@@ -1702,6 +1702,7 @@ html_sink_cc_tests ()
 }
 
 } // namespace selftest
-} // namespace diagnostics
 
 #endif /* CHECKING_P */
+
+} // namespace diagnostics
diff --git a/gcc/diagnostics/output-spec.cc b/gcc/diagnostics/output-spec.cc
index 08128a92add1..83f128cc536b 100644
--- a/gcc/diagnostics/output-spec.cc
+++ b/gcc/diagnostics/output-spec.cc
@@ -846,6 +846,7 @@ output_spec_cc_tests ()
 }
 
 } // namespace diagnostics::selftest
-} // namespace diagnostics
 
 #endif /* #if CHECKING_P */
+
+} // namespace diagnostics
diff --git a/gcc/diagnostics/sarif-sink.cc b/gcc/diagnostics/sarif-sink.cc
index 05c0a8eb3a92..4738ae9c3007 100644
--- a/gcc/diagnostics/sarif-sink.cc
+++ b/gcc/diagnostics/sarif-sink.cc
@@ -5072,6 +5072,7 @@ sarif_sink_cc_tests ()
 }
 
 } // namespace diagnostics::selftest
-} // namespace diagnostics
 
 #endif /* CHECKING_P */
+
+} // namespace diagnostics
-- 
2.43.7

